packages/cinterop/src/darwin/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -1460,75 +1482,57 @@ actual object RealmInterop {
-                printlntid("$scheduler notify")
-                try {
-                    scheduler.notify()
-                } catch (e: Exception) {
-                    // Should never happen, but is included for development to get some indicators
-                    // on errors instead of silent crashes.
-                    e.printStackTrace()
-                }
-            },
-
-            // is_on_thread: realm_wrapper.realm_scheduler_is_on_thread_func_t? /* = kotlinx.cinterop.CPointer<kotlinx.cinterop.CFunction<(kotlinx.cinterop.COpaquePointer? /* = kotlinx.cinterop.CPointer<out kotlinx.cinterop.CPointed>? */) -> kotlin.Boolean>>? */,
-            staticCFunction<COpaquePointer?, Boolean> { userdata ->
-                // Must be thread safe
-                val scheduler =
-                    userdata!!.asStableRef<SingleThreadDispatcherScheduler>().get()
-                printlntid("is_on_thread[$scheduler] ${scheduler.threadId} " + tid())
-                scheduler.threadId == tid()
-            },
-
-            // is_same_as: realm_wrapper.realm_scheduler_is_same_as_func_t? /* = kotlinx.cinterop.CPointer<kotlinx.cinterop.CFunction<(kotlinx.cinterop.COpaquePointer? /* = kotlinx.cinterop.CPointer<out kotlinx.cinterop.CPointed>? */, kotlinx.cinterop.COpaquePointer? /* = kotlinx.cinterop.CPointer<out kotlinx.cinterop.CPointed>? */) -> kotlin.Boolean>>? */,
-            staticCFunction<COpaquePointer?, COpaquePointer?, Boolean> { userdata, other ->
-                userdata == other
-            },
+                // free: realm_wrapper.realm_free_userdata_func_t? /* = kotlinx.cinterop.CPointer<kotlinx.cinterop.CFunction<(kotlinx.cinterop.COpaquePointer? /* = kotlinx.cinterop.CPointer<out kotlinx.cinterop.CPointed>? */) -> kotlin.Unit>>? */,
+                staticCFunction<COpaquePointer?, Unit> { userdata ->
+                    printlntid("free")
+                    userdata?.asStableRef<SingleThreadDispatcherScheduler>()?.dispose()
+                },
 
-            // can_deliver_notifications: realm_wrapper.realm_scheduler_can_deliver_notifications_func_t? /* = kotlinx.cinterop.CPointer<kotlinx.cinterop.CFunction<(kotlinx.cinterop.COpaquePointer? /* = kotlinx.cinterop.CPointer<out kotlinx.cinterop.CPointed>? */) -> kotlin.Boolean>>? */,
-            staticCFunction<COpaquePointer?, Boolean> { userdata -> true },
+                // notify: realm_wrapper.realm_scheduler_notify_func_t? /* = kotlinx.cinterop.CPointer<kotlinx.cinterop.CFunction<(kotlinx.cinterop.COpaquePointer? /* = kotlinx.cinterop.CPointer<out kotlinx.cinterop.CPointed>? */) -> kotlin.Unit>>? */,
+                staticCFunction<COpaquePointer?, Unit> { userdata ->
+                    // Must be thread safe
+                    val scheduler =
+                        userdata!!.asStableRef<SingleThreadDispatcherScheduler>().get()
+                    printlntid("$scheduler notify")
```

Comment:
Just pops up here because of initially basing this on the key-caching PR, so nothing really changed. I will leave it as it. It is a no-op as `printlntid` is an inline no-op method that can be redefined for debugging purpose.